tiledb/sm/misc/parallel_functions.h

```diff
@@ -270,9 +270,7 @@ Status parallel_for(
   }
 
   // Wait for all instances of `execute_subrange` to complete.
-  // This is ignoring the wait status as we use failed_exception for propagating
-  // the tasks exceptions.
-  std::ignore = tp->wait_all(tasks);
+  throw_if_not_ok(tp->wait_all(tasks));
```

Comment:
Is the `failed_exception` block dead code now?

Reply:
Yes, I am also removing this code and refactoring the error handling in https://github.com/TileDB-Inc/TileDB/pull/3601

Reply:
I will write a story shortly about updating `ThreadPool` to combine all exceptions it receives in `wait_all` and to throw a compound exception.  A question is -- where should that be caught/